### PR TITLE
chore: add GitHub action to run tests

### PR DESCRIPTION
**Pull request by targos**
Let's see how it goes.

### CODE REVIEW

**Comment by targos** [comment]
It looks like the `run` syntax I used doesn't work on Windows (only `npm install` is executed, not `npm test`).

But this is how GitHub documents it:
image

Any idea?

**Comment by targos** [comment]
The bug was reported: https://github.community/t5/GitHub-Actions/Windows-multi-line-step-run/m-p/30428#M426

**Comment by targos** [comment]
It works fine with `pwsh`.

**Comment by richardlau** [comment]
I'm up for giving Actions a try. Travis is currently perma-red due to https://github.com/nodejs/citgm/issues/747 (which I haven't been able to replicate locally and it doesn't appear to have happened in the Actions runs). 

Curious about the extra permissions for codecov... we can't call the bash script like we already do in Travis? https://github.com/nodejs/citgm/blob/2204de9d9549728774c2a1cc4ad07e8ad4b27248/.travis.yml#L12

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: Node CI
```
Comment (richardlau) [inline .github/workflows/nodejs.yml]:
```suggestion
name: Node.js CI
```

**Comment by targos** [comment]
> I'm up for giving Actions a try. Travis is currently perma-red due to #747 (which I haven't been able to replicate locally and it doesn't appear to have happened in the Actions runs).
> 
> Curious about the extra permissions for codecov... we can't call the bash script like we already do in Travis?
> 
> https://github.com/nodejs/citgm/blob/2204de9d9549728774c2a1cc4ad07e8ad4b27248/.travis.yml#L12

We can call the bash script, but unlike with Travis, we have to provide it a repository token that codecov generates, and to get the token, we need to allow Codecov to access the org.
See this action that I intend to use: https://github.com/codecov/codecov-action

**Comment by targos** [comment]
I pushed a new commit to propose a compromise. Now GitHub actions run the full matrix and Travis only runs on Node.js 12. I kept Travis because it's simpler for Codecov and works with PRs opened from forks.

**Comment by targos** [comment]
@nodejs/citgm

**Comment by richardlau** [comment]
> 
> 
> I pushed a new commit to propose a compromise. Now GitHub actions run the full matrix and Travis only runs on Node.js 12. I kept Travis for Codecov because it's simpler and works with PRs opened from forks.

Actions don't run on PR's from forks?

We could make Travis run on lts and current (and thus avoid having to periodically update the yaml for new versions of Node.js). Otherwise if we're fixing a version maybe fix at the earliest still supported version of Node.js (currently 8)?

**Comment by targos** [comment]
Actions do run on PRs from forks, but they don't have access to secrets so codecov couldn't work

**Comment by targos** [comment]
I moved the linter to a separate job

**Comment by targos** [comment]
I'd like to have another review from the team. /cc @BridgeAR @MylesBorins

File: .github/workflows/nodejs.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+name: Node.js CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
```
Comment (BridgeAR) [inline .github/workflows/nodejs.yml]: Does this work like a wildcard? I can not find anything about this in the action documentation.

Comment (targos) [reply .github/workflows/nodejs.yml]: I don't understand the question

Comment (BridgeAR) [reply .github/workflows/nodejs.yml]: The action is defined to be triggered by pull requests, but it's not specified by what pull requests. I guess by just adding the category without any branches it works like:

```
  pull_request:
    branches:
      - *
```

Comment (targos) [reply .github/workflows/nodejs.yml]: yes, I guess it's the same. I used it in other projects and it triggers on every pull request

**Pull request merged**